src/accountsdb/db.zig

```diff
@@ -1434,18 +1484,17 @@ pub const AccountsDB = struct {
         defer cleaned_pubkeys.deinit();
 
         for (unclean_account_files) |file_id| {
-            var account_file_rw = blk: {
-                const file_map, var file_map_lg = self.file_map.readWithLock();
-                defer file_map_lg.unlock();
-
-                break :blk file_map.get(file_id).?;
-            };
-
             // NOTE: this read-lock is held for a while but
             // is not expensive since writes only happen
             // during shrink/delete (which dont happen in parallel to this fcn)
-            const account_file, var account_file_lg = account_file_rw.readWithLock();
-            defer account_file_lg.unlock();
+            self.file_map_safe.fd_keep_open_rw.lockShared();
+            defer self.file_map_safe.fd_keep_open_rw.unlockShared();
+
+            const account_file = blk: {
+                const file_map, var file_map_lg = self.file_map_safe.file_map.readWithLock();
+                defer file_map_lg.unlock();
+                break :blk file_map.get(file_id).?;
+            };
```

Comment:
this is a pretty common pattern which i feel like we should be able to contain in a method on self.file_map_safe - wdyt

my main concern is i dont like how many field accesses we need everywhere (ie, `self.file_map_safe.fd_keep_open_rw`) - should be a better way